packages/wallet/src/redux/protocols/challenging/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,214 @@
+import * as scenarios from './scenarios';
+import { challengingReducer, initialize, ReturnVal } from '../reducer';
+import { FailureReason } from '../states';
+
+describe('opponent-responds scenario', () => {
+  const scenario = scenarios.opponentResponds;
+  const { channelId, processId, storage } = scenario;
+
+  describe('when initializing', () => {
+    const result = initialize(channelId, processId, storage);
+
+    itTransitionsTo(result, 'ApproveChallenge');
```

Comment:
Can we make these type-safe in some way?

 Otherwise if we follow this pattern we're going to have to be very vigilant on typos and refactoring.

Reply:
Done with `export type ChallengingStateType = ChallengingState['type'];`.

(As it happens, I don't think the problem is actually that bad here anyhow, as any typo will be immediately caught by the test. The other case we use these strings is in discriminated unions and in direct comparisons with `state.type` - both of which will be picked up by the type checker.)